### ISSUE TITLE
Flask-Secirity allows scanning emails to find out if a user is a member

### ISSUE THREAD

**Issue by jimbojetlag** [mattupstate/flask-security#476]
With default settings, if you provide:

1. an email address which is not in the database
2. a wrong password

you'll get the error: Specified user does not exist

Also, if you provide:

1. an email address which *is* in the database
2. a wrong password

you'll get the error: Invalid password

Both of these methods opens vulnerability to scan the user database for a given email. This is bad practice for both security and privacy.

The error for both of the above cases should be: Invalid email or password.

And, this should be the default.

**Comment by jonafato** [comment]
Agreed that Flask-Security should not leak this information. Open to a pull request with better defaults.

**Comment by DeimosA** [comment]
You can manually set the messages to be the same, but i agree that this should be the default

**Comment by Avamander** [comment]
This can be solved by overriding the error messages, adding a captcha and rate-limiting to pretty much all the account-related built-in pages, last two you should be doing anyways... I think this should be done by the developer, implementing captchas and rate limiting on Flask-Security's side sounds like feature creep.

**Comment by wjurkowlaniec** [comment]
also in forgot password view it's possible
